nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -1112,6 +1121,11 @@ private void setHeaderProperties(final ProcessContext context, final Request.Bui
             requestBuilder.addHeader("Date", RFC_2616_DATE_TIME.format(universalCoordinatedTimeNow));
         }
 
+        if (context.getProperty(OAUTH2_ACCESS_TOKEN_PROVIDER).isSet()) {
+            OAuth2AccessTokenProvider oauth2AccessTokenProvider = context.getProperty(OAUTH2_ACCESS_TOKEN_PROVIDER).asControllerService(OAuth2AccessTokenProvider.class);
+            requestBuilder.addHeader("Authorization", "Bearer " + oauth2AccessTokenProvider.getAccessToken());
```

Comment:
Setting the `Authorization` header will conflict with the Username and Password properties used for configuring Basic or Digest authentication. One way to address this is introducing more logic in `customValidate()` to ensure that those properties are not set when OAuth2 Access Token Provider is set.